### ISSUE TITLE
The Sync sign-in success page just says "Sign in to this Firefox to complete set-up" without any other context (if you're not signed in)

### ISSUE THREAD

**Issue by dholbert** [mozilla/fxa#16053]
## Description

tl;dr: If you're not signed in to Sync or Accounts, https://accounts.firefox.com/connect_another_device is a little too mysterious -- it just says "Sign in to this Firefox to complete set-up", without providing any other info about what it's doing or what it's for.

This matters because a user might have a tab or window open at this URL, from a previous trial-run of Sync which they decided against using. (The sync sign-in process terminates at this URL.)  If a user stumbles back across this page with its lack-of-information about why it's there or what's going to happen, users might perceive this mysterious "Sign in to complete set-up" page as a dark pattern, trying to trick them into giving Sync another try when they already chose not to use it.

### Steps to reproduce
(1) User signs into Firefox Sync (maybe clicking through some Firefox UI to create/sign-in to an account for the first time).
(2) At the end of the Sync sign-in/setup process, the user lands at a final URL of something like https://accounts.firefox.com/connect_another_device?context=fx_desktop_v3&entrypoint=preferences&action=email&service=sync
(3) Let's say the user leaves that Firefox window alone and forgets about it, and then decides they don't want to use Sync after all. So, they sign out of their browser (e.g. from the Sync toolbar icon or from preferences), and they sign out of https://accounts.firefox.com.
(4) Later on, the user quits Firefox and restores their previously-opened windows.

### Actual result
The user ends up with a restored Firefox window viewing a page that they've never seen before, which just says the following:
```
Sign in to this Firefox to complete set-up
[Sign in]
```
...with no other context.

They've got this page because it's the URL that was left behind from the Sync sign-in process (step 2) and this happens to be how that URL renders if you're not signed in. The user then potentially gets frustrated with Firefox because (from their perspective) it looks like we're annoying them to sign back into Sync when they already decided against it.

### Expected result
The Sync setup process final-page should probably be designed to be something with a more "inert" rendering if you happen to still have it around after having logged out of Sync (as in the STR).   Or, if we really want it to have a sign-in button, we should provide clearer context about what set-up we're asking the user to complete by clicking through there.

For context, I'm spinning this off of a bug report from https://bugzilla.mozilla.org/show_bug.cgi?id=1833050 , where a user got very frustrated thinking Sync was repeatedly trying to nudge them to sign in. I'm not sure my STR exactly match what the user did on that bug, but this is the closest I could come to reconstructing what might have happened.

### Environment
Firefox Nightly 121.0a1 on Ubuntu 22.04

**Comment by dholbert** [comment]
Here's a screencast showing how a user might arrive at this page unexpectedly:
[Screencast from 11-07-2023 09:36:42 PM.webm](https://github.com/mozilla/fxa/assets/426803/c1558fed-5357-4836-995d-ddaa070ddb2a)

(This is essentially following the STR from the initial comment here.)

**Comment by dholbert** [comment]
Here's a screenshot of how the page-in-question[1] looks under the conditions-in-question[2].  The user-scenario here is a user stumbling on this page unexpectedly after restarting Firefox, without any recollection of seeing it before:
![image](https://github.com/mozilla/fxa/assets/426803/35b05ddb-0009-4d29-b846-5a57d64d3986)

[1] page: https://accounts.firefox.com/connect_another_device
[2] conditions: when not signed in to Sync or to https://accounts.firefox.com/ (e.g. after clicking "sign out" for both in a separate window and restarting Firefox, as shown in the screencast).

**Comment by dholbert** [comment]
> For context, I'm spinning this off of a bug report from https://bugzilla.mozilla.org/show_bug.cgi?id=1833050 , where a user got very frustrated thinking Sync was repeatedly trying to nudge them to sign in. I'm not sure my STR exactly match what the user did on that bug, but this is the closest I could come to reconstructing what might have happened.

Update: we discovered today that the user's frustration actually had nothing to do with what I described here; it turns out they were running into a rough edge between Sync and the Firefox "Primary Password" feature.

So: maybe this isn't a big deal. I do still think this page could be made a bit less vague about what's going to happen and why you're seeing it, as described in my initial comment here, but it's admittedly a bit of a contrived setup where a user might actually be upset by the actual-results here.
